### DISCUSSION TITLE
simplifying-theory-test-data-with-xunit-combinatorial/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#431] [Announcements]
# simplifying-theory-test-data-with-xunit-combinatorial/

In this post I show how you can simplify your xUnit [Theory] tests using the Xunit.Combinatorial package to auto-generate your test parameters

https://andrewlock.net/simplifying-theory-test-data-with-xunit-combinatorial/

**Comment by EngincanV** [comment]
Thanks for the post, Andrew. It seems really helpful to tidy up methods that run against multiple values and combinations. Can't wait to refactor my tests!

**Comment by lkurzyniec** [comment]
I bet you already know AutoFixture’s `AutoData` attribute. It's similar, but different ;-) As well useful to reduce amount of code.
